packages/agents/router/src/lib/operations/execute.ts

```diff
@@ -75,30 +71,41 @@ export const execute = async (params: OriginTransfer): Promise<void> => {
   // based on a calculation of which rounds we can afford to bid on. For now, this is hardcoded to bid
   // only on the first auction round.
   // Produce the router path signatures for each auction round we want to bid on.
-  const signatures = {
-    "1": await signRouterPathPayload(transferId, "1", wallet),
-  };
-
-  logger.debug("Signed payloads", requestContext, methodContext, {
-    rounds: Object.keys(signatures),
-    // Sanitized with ellipsis.
-    sigs: Object.values(signatures).map((s) => s.slice(0, 6) + ".."),
-  });
 
-  // sanity check
+  // Make a list of signatures that reflect which auction rounds we want to bid on.
   const balance = await subgraph.getAssetBalance(destinationDomain, routerAddress, executeLocalAsset);
-  if (balance.lt(receivingAmount)) {
+  const signatures: Record<string, string> = {};
+  for (let roundIdx = 1; roundIdx <= config.maxPathLength; roundIdx++) {
+    const amountForRound = getAuctionAmount(roundIdx, BigNumber.from(receivingAmount));
+    if (amountForRound.lte(balance)) {
+      signatures[roundIdx.toString()] = await signRouterPathPayload(transferId, roundIdx.toString(), wallet);
```

Comment:
this won't work. Routers can't sign bids with the round number - they need to sign the path length!

So for round 3, if the transfer amount is split 4-ways, the router should be signing "4" for the path length, not "3"!
```ts
signRouterPathPayload(transferId, "4", wallet);
```

Reply:
> this won't work. Routers can't sign bids with the round number - they need to sign the path length!
> 
> So for round 3, if the transfer amount is split 4-ways, the router should be signing "4" for the path length, not "3"!
> 
> ```ts
> signRouterPathPayload(transferId, "4", wallet);
> ```

good catch!

Reply:
bump. it looks like we're still passing in roundIdx.toString() ? We should be translating roundIdx => pathLength